drivers/network/dd/e1000/hardware.c

```diff
@@ -549,9 +591,13 @@ NICPowerOn(
         return Status;
     }
 
-    if (!E1000ValidateNvmChecksum(Adapter))
+    Adapter->E1000_EERD_ADDR_SHIFT = 0xffff;
+    if (E1000DetermineAndSaveEERDLayout(Adapter))
```

Comment:
You're returning success if this fails. That doesn't seem right.

Reply:
Yes and No ;-) I want to only validate the EEPROM if it exists (The documentation says "the only way to see if EEPROM is available is to read it") In the PermanentMAC function I try to read the MAC from EEPROM only if the EEPROM is valid.